ncm-metaconfig/src/main/metaconfig/singularity/pan/schema.pan

```diff
@@ -0,0 +1,118 @@
+declaration template metaconfig/singularity/schema;
+
+include 'pan/types';
+
+type directory = string with match(SELF,'[^/]+/?$');
```

Comment:
do not define these generic types here. should be in `template-library-core` or just use the type `absolute_file_path` instead